model/modelpb/apmevent.pb.json.go

```diff
@@ -0,0 +1,279 @@
+		doc.DataStreamNamespace = e.DataStream.Namespace
+	}
+
+	if e.Processor != nil {
+		doc.Processor = modeljson.Processor{
+			Name:  e.Processor.Name,
+			Event: e.Processor.Event,
+		}
+	}
+
+	if e.Transaction != nil {
+		e.Transaction.toModelJSON(doc.Transaction, e.Processor.Name == "metric" && e.Processor.Event == "metric")
+	}
+
+	if e.Span != nil {
+		e.Span.toModelJSON(doc.Span)
+	}
+
+	if e.Metricset != nil {
+		e.Metricset.toModelJSON(doc.Metricset)
+		doc.DocCount = e.Metricset.DocCount
+	}
+
+	if e.Error != nil {
+		e.Error.toModelJSON(doc.Error)
+	}
+
+	if e.Event != nil {
+		e.Event.toModelJSON(doc.Event)
+	}
+
+	// Set high resolution timestamp.
+	//
+	// TODO(axw) change @timestamp to use date_nanos, and remove this field.
+	var timestampStruct modeljson.Timestamp
+	if !e.Timestamp.AsTime().IsZero() {
+		if e.Processor != nil {
+			processorName := e.Processor.Name
+			processorEvent := e.Processor.Event
+			if (processorName == "error" && processorEvent == "error") || (processorName == "transaction" && (processorEvent == "transaction" || processorEvent == "span")) {
+				timestampStruct.US = int(e.Timestamp.AsTime().UnixNano() / 1000)
+				doc.TimestampStruct = &timestampStruct
+			}
+		}
+	}
+
+	if e.Cloud != nil {
+		e.Cloud.toModelJSON(doc.Cloud)
+	}
+
+	if e.Faas != nil {
+		e.Faas.toModelJSON(doc.FAAS)
+	}
+
+	if e.Device != nil {
+		e.Device.toModelJSON(doc.Device)
+	}
+
+	if e.Network != nil {
+		e.Network.toModelJSON(doc.Network)
+	}
+
+	if e.Observer != nil {
+		e.Observer.toModelJSON(doc.Observer)
+	}
+
+	if e.Container != nil {
+		e.Container.toModelJSON(doc.Container)
+	}
+
+	if e.Kubernetes != nil {
+		e.Kubernetes.toModelJSON(doc.Kubernetes)
+	}
+
+	if e.Agent != nil {
+		e.Agent.toModelJSON(doc.Agent)
+	}
+
+	if e.Trace != nil {
+		doc.Trace.ID = e.Trace.Id
+	}
+
+	if e.User != nil {
+		e.User.toModelJSON(doc.User)
+	}
+
+	if e.Source != nil {
+		e.Source.toModelJSON(doc.Source)
+	}
+
+	if e.Parent != nil {
+		doc.Parent.ID = e.Parent.Id
+	}
+
+	if e.Child != nil {
+		doc.Child.ID = e.Child.Id
+	}
+
+	if e.Client != nil {
+		e.Client.toModelJSON(doc.Client)
+	}
+
+	if e.UserAgent != nil {
+		doc.UserAgent.Original = e.UserAgent.Original
+		doc.UserAgent.Name = e.UserAgent.Name
+	}
+
+	if e.Service != nil {
+		e.Service.toModelJSON(doc.Service)
+	}
+
+	if e.Http != nil {
+		e.Http.toModelJSON(doc.HTTP)
+	}
+
+	if e.Host != nil {
+		e.Host.toModelJSON(doc.Host)
+	}
+
+	if e.Url != nil {
+		e.Url.toModelJSON(doc.URL)
+	}
+
+	if e.Log != nil {
+		e.Log.toModelJSON(doc.Log)
+	}
+
+	if e.Process != nil {
+		e.Process.toModelJSON(doc.Process)
+	}
+
+	if e.Destination != nil {
+		e.Destination.toModelJSON(doc.Destination)
+	}
+
+	if e.Session != nil {
+		doc.Session.ID = e.Session.Id
+		doc.Session.Sequence = int(e.Session.Sequence)
+	}
+}
+
+func setNonZero[T comparable](to **T, from *T) {
+	if !isZero(*from) {
+		*to = from
+	}
+}
+
+func isZero[T comparable](t T) bool {
+	var zero T
+	return t == zero
+}
```

Comment:
```suggestion
```

These don't seem to be used, can we remove them?

Reply:
Good catch! Removed!